src/app/stack/card-details/component-information/component-information.component.ts

```diff
@@ -40,14 +42,26 @@ export class ComponentInformationComponent implements OnInit, OnChanges {
     ngOnChanges(changes: SimpleChanges) {
         let summary: any = changes['component'];
         if (summary) {
-            this.component = <MComponentInformation | MRecommendationInformation> summary.currentValue;
+            this.component = <MComponentInformation | MRecommendationInformation>summary.currentValue;
         }
         if (changes['positions']) {
             this.positions = changes['positions']['currentValue'];
         }
+
         this.paint();
     }
 
+    public generateUrl(url: string): string {
+        if (typeof (url) === "string") {
+            let content: Array<string>;
+            let generatedUrl: string;
+            content = url.split('/vuln/', 2);
+            generatedUrl = url + "?utm_medium=Partner&utm_source=RedHat&utm_campaign=Code-Ready-Analytics-2020&utm_content=vuln/" + content[1];
```

Comment:
Can you create a separate file to maintain utm content instead of spreading this across codebase?

Reply:
fixed